tools/analysis_tools/eval_metric.py

```diff
@@ -35,53 +24,28 @@ def parse_args():
         'It also allows nested list/tuple values, e.g. key="[(a,b),(c,d)]" '
         'Note that the quotation marks are necessary and that no white space '
         'is allowed.')
-    parser.add_argument(
-        '--eval-options',
-        nargs='+',
-        action=DictAction,
-        help='custom options for evaluation, the key-value pair in xxx=yyy '
-        'format will be kwargs for dataset.evaluate() function')
     args = parser.parse_args()
     return args
 
 
 def main():
     args = parse_args()
 
-    cfg = Config.fromfile(args.config)
-
-    # replace the ${key} with the value of cfg.key
-    cfg = replace_cfg_vals(cfg)
+    register_all_modules(init_default_scope=True)
 
-    # update data root according to MMDET_DATASETS
-    update_data_root(cfg)
-
-    assert args.eval or args.format_only, (
-        'Please specify at least one operation (eval/format the results) with '
-        'the argument "--eval", "--format-only"')
-    if args.eval and args.format_only:
-        raise ValueError('--eval and --format_only cannot be both specified')
+    cfg = Config.fromfile(args.config)
 
     if args.cfg_options is not None:
         cfg.merge_from_dict(args.cfg_options)
-    cfg.data.test.test_mode = True
 
-    dataset = build_dataset(cfg.data.test)
-    outputs = load(args.pkl_results)
+    dataset = DATASETS.build(cfg.test_dataloader.dataset)
+    predictions = mmengine.load(args.pkl_results)
+    assert len(dataset) == len(predictions)
```

Comment:
this line seems useless because we no longer need to pass the dataset to the evaluator